buildmachine/install-load-tests.ps1

```diff
@@ -0,0 +1,100 @@
+#Requires -RunAsAdministrator
+
+[CmdletBinding()]
+param (
+    [Parameter()][string]$JMeterVersion = "5.4.1",
+    [Parameter()][string]$DownloadFolder = "C:\download",
+    [Parameter()][string]$JMeterPath = "C:\tools\jmeter",
+    [Parameter()][string]$PluginsList = "jpgc-graphs-basic,jpgc-casutg,jpgc-prmctl"
+)
+
+function ExtractArchive([string]$archiveFile, [string]$outputFolder) {
+    Write-Host "    Extract archive '$archiveFile' to '$outputFolder'" -ForegroundColor Blue
+
+    $extractCommand = "7z x $archiveFile -o$jmeterSource -r -aoa"
+    Invoke-Expression $extractCommand
+}
+
+function AddToPath([string]$path) {
+	Write-Host "  Add '$path' to environment variable PATH" -ForegroundColor Yellow
+	$pathValue = [System.Environment]::GetEnvironmentVariable("PATH", [System.EnvironmentVariableTarget]::Machine)
+	$pattern = $path.Replace("\", "\\")
+	if ($pathValue -match $pattern) {
+		Write-Host "    PATH variable already contains '$path'."
+	}
+	else {
+		$pathValue = "$path;" + $pathValue
+		[System.Environment]::SetEnvironmentVariable("PATH", $pathValue, [System.EnvironmentVariableTarget]::Machine)
+		Write-Host "    '$path' added to PATH variable."
+	}
+}
+
+# Choco install
+Write-Host "Install software for load testing" -ForegroundColor Green
+choco install load-tests-buildmachine-packages.config --yes
+
+# Add JAVA folder to PATH
+Write-Host "Add JAVA folder to PATH" -ForegroundColor Green
+$javaHomeValue = [System.Environment]::GetEnvironmentVariable("JAVA_HOME", [System.EnvironmentVariableTarget]::Machine)
+if ([string]::IsNullOrEmpty($javaHomeValue)) {
+    Write-Host "    'JAVA_HOME' is not set, so Java variables will not be set ('JAVA', 'JAVA_FLAGS', 'SONAR_SCANNER_OPTS')."
+}
+else {
+    $javaBinPath = Join-Path -Path $javaHomeValue -ChildPath "bin"
+    AddToPath $javaBinPath
+}
+
+# Install JMeter
+Write-Host "Install JMeter" -ForegroundColor Green
+
+$jmeterSource = Join-Path -Path $DownloadFolder -ChildPath "jmeter"
+$jmeterZipFile = Join-Path -Path $jmeterSource -ChildPath "jmeter.tgz"
+$tarFile = Join-Path -Path $jmeterSource -ChildPath "jmeter.tar"
+$extractedFolder = Join-Path -Path $jmeterSource -ChildPath "apache-jmeter-$JMeterVersion"
+$pluginManagerFile = Join-Path -Path $JMeterPath -ChildPath "lib/ext/jmeter-plugins-manage.jar"
+$cmdRunnerFile = Join-Path -Path $JMeterPath -ChildPath "lib/cmdrunner-2.2.jar"
+
+## Delete source and target folder
+Write-Host "  Delete JMeter folder '$JMeterPath'" -ForegroundColor Yellow
+Remove-Item $JMeterPath -Recurse -ErrorAction Ignore
+Remove-Item $jmeterSource -Recurse -ErrorAction Ignore
+
+## Download JMeter
+Write-Host "  Download JMeter (v $JMeterVersion)" -ForegroundColor Yellow
+Invoke-WebRequest -Uri "https://downloads.apache.org//jmeter/binaries/apache-jmeter-$JMeterVersion.tgz" -OutFile ( New-Item -Path $jmeterZipFile -Force)
```

Comment:
Načo je tu `New-Item`? Ako hodnota pre `-OutFile` stačí cesta (string) a on ten súbor vytvorí. Ak súbor existuje, nebabre sa s ním a rovno ho prepíše.  Takto to máš pri všetkých `Invoke-WebRequest`.

Reply:
Si si istý, mne stále padal na tom, že cieľový súbor existuje. Preto som to robil takto. Ale môžem opäť overiť.

Reply:
Hm. Ja som si to vyskúšal v robote. Tak neviem či to ešte závisí od verzie Powershellu alebo čo.

Reply:
Už si spomínam prečo som to tam dával. `Invoke-WebRequest` zlyhá, keď neexistuje časť cesty kam to chceš uložiť. _(V mojom prípade adresár `c:\donwload`)_. Ale keďže to teraz sťahujem do `$env:Temp` tak to už nieje potrebné. Odstránim.